Integrations/integration-MaxMind_GeoIP2.yml

```diff
@@ -0,0 +1,289 @@
+  name: url
+  defaultvalue: https://geoip.maxmind.com/
+  type: 0
+  required: true
+script:
+  script: |
+    import requests
+    from collections import defaultdict
+    from requests.auth import HTTPBasicAuth
+    # disable insecure warnings
+    requests.packages.urllib3.disable_warnings()
+
+
+    '''GLOBAL VARS'''
+    BASE_URL = demisto.params().get('url')
+    APIKEY = demisto.params().get('apikey')
+    ACCOUNT_ID = demisto.params().get('account')
+    MODE = demisto.params().get('mode')
+    INSECURE = demisto.params().get('insecure')
+    PROXY = demisto.params().get('proxy')
+    API_VERSION = 'geoip/v2.1'
+
+    if not demisto.params().get('proxy', False):
+        del os.environ['HTTP_PROXY']
+        del os.environ['HTTPS_PROXY']
+        del os.environ['http_proxy']
+        del os.environ['https_proxy']
+
+    HR_HEADERS = [
+        'IP',
+        'Domain',
+        'ASN',
+        'Organization',
+        'ISP',
+        'Location',
+        'Accuracy Radius',
+        'User Type',
+        'Continent',
+        'ISO Code',
+        'Country',
+        'Registered Country',
+        'TimeZone',
+        'City',
+        'Subdivision',
+        'Is TOR Exit Node',
+        'Is Hosting Provider',
+        'Is Anonymous']
+
+    HEADERS = {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json'
+    }
+
+
+    '''HELPER FUNCTIONS'''
+    def http_request(query):
+        r = requests.request(
+            'GET',
+            BASE_URL + API_VERSION + '/' + MODE +'/'+ query,
+            headers=HEADERS,
+            verify=INSECURE,
+            auth=HTTPBasicAuth(ACCOUNT_ID, APIKEY)
+        )
+        if r.status_code is not 200:
+            return_error('Error in API call to MaxMind, got status code - {} and a reason: {}'.format(r.status_code, r.reason))
+        return r
+
+
+    def format_results(res_json):
+        hr = defaultdict()
+        maxmind_ec = defaultdict(lambda: defaultdict(int))
+        ip_ec = defaultdict(lambda: defaultdict(int))
+        if 'continent' in res_json:
+            continent = res_json['continent']
+            hr['Continent'] = continent['names']['en']
+            maxmind_ec['Geo']['Continent'] = continent['names']['en']
+        if 'city' in res_json:
+            city = res_json['city']
+            hr['City'] = city['names']['en']
+            maxmind_ec['Geo']['City'] = city['names']['en']
+        if 'country' in res_json:
+            country = res_json['country']
+            hr['Country'] = country['names']['en']
+            maxmind_ec['Geo']['Country'] = country['names']['en']
+            ip_ec['Geo']['Country'] = country['names']['en']
+        if 'location' in res_json:
+            location = res_json['location']
+            ip_ec['Geo']['Location'] = str(location['latitude'])+', '+str(location['longitude'])
+            maxmind_ec['Geo']['Location'] = str(location['latitude'])+', '+str(location['longitude'])
+            demisto.results({
```

Comment:
lets extract this to another function, it took me a sec to realise what are you doing here so better to call the function `create_map_entry` or something of that sort.